src/store/modules/ProfileExportModule.ts

```diff
@@ -0,0 +1,69 @@
+import R2Error from '../../model/errors/R2Error';
+import ProfileModList from '../../r2mm/mods/ProfileModList';
+import InteractionProvider from '../../providers/ror2/system/InteractionProvider';
+import { ActionTree, MutationTree } from 'vuex';
+import { State as RootState } from '../../store';
+
+interface State {
+    exportCode?: string;
+}
+
+export default {
+    namespaced: true,
+
+    state: (): State => ({
+        exportCode: undefined,
+    }),
+
+    mutations: <MutationTree<State>>{
+        reset(state: State) {
+            state.exportCode = undefined;
+        },
+
+        setExportCode(state: State, code: string) {
+            state.exportCode = code;
+        },
+    },
+
+    actions: <ActionTree<State, RootState>>{
+
+        async exportProfileAsFile({rootGetters}) {
+            if (!rootGetters["profile/modList"].length) {
+                throw new R2Error(
+                    'Profile is empty',
+                    'The profile must contain at least one mod to export it as a file.'
+                );
+            }
+            const immutableProfile = rootGetters["profile/activeProfileOrThrow"].asImmutableProfile();
+            const exportErr = await ProfileModList.exportModListToFile(immutableProfile);
```

Comment:
FYI I've added [throwForR2Error](https://github.com/ebkr/r2modmanPlus/blob/develop/src/model/errors/R2Error.ts#L34) that can be used in situations like this.